pylib/mozhginfo/mozhginfo/pushlog_client.py

```diff
@@ -0,0 +1,187 @@
+#! /usr/bin/env python
+# coding: utf-8
+"""
+This helps us query information about Mozilla's Mercurial repositories.
+
+Documentation found in here:
+http://mozilla-version-control-tools.readthedocs.org/en/latest/hgmo/pushlog.html
+
+Important notes from the pushlog documentation::
+
+    When implementing agents that consume pushlog data, please keep in mind
+    the following best practices:
+
+    * Query by push ID, not by changeset or date.
+    * Always specify a startID and endID.
+    * Try to avoid full if possible.
+    * Always use the latest format version.
+    * Don't be afraid to ask for a new pushlog feature to make your life easier.
+"""
+import logging
+
+import requests
+
+
+LOG = logging.getLogger('pushlog_client')
+JSON_PUSHES = "%(repo_url)s/json-pushes"
+VALID_CACHE = {}
+
+
+class PushlogError(Exception):
+    pass
+
+
+def query_revisions_range(repo_url, from_revision, to_revision, version=2, tipsonly=1):
+    """
+    Return an ordered list of revisions (by date - oldest (starting) first).
+
+    repo           - represents the URL to clone a repo
+    from_revision - from which revision to start with (oldest)
+    to_revision   - from which revision to end with (newest)
+    version        - version of json-pushes to use (see docs)
+    """
+    revisions = []
+    url = "%s?fromchange=%s&tochange=%s&version=%d&tipsonly=%d" % (
+        JSON_PUSHES % {"repo_url": repo_url},
+        from_revision,
+        to_revision,
+        version,
+        tipsonly
+    )
+    LOG.debug("About to fetch %s" % url)
+    req = requests.get(url)
+    pushes = req.json()["pushes"]
+    # json-pushes does not include the starting revision
+    revisions.append(from_revision)
+    for push_id in sorted(pushes.keys()):
+        # Querying by push ID is perferred because date ordering is
+        # not guaranteed (due to system clock skew)
+        # We can interact with self-serve with the full char representation
+        revisions.append(pushes[push_id]["changesets"][-1])
+
+    return revisions
+
+
+def query_pushid_range(repo_url, start_id, end_id, version=2):
+    """
+    Return an ordered list of revisions (newest push id first).
```

Comment:
It feels weird that the above method sorts oldest to newest and this one is newest to oldest. I'd be consistent. Or, throw in an argument to control sort order.